pkg/runtime/optimize/optimizer.go

```diff
@@ -247,6 +247,20 @@ func (o optimizer) optimizeSelect(ctx context.Context, conn proto.VConn, stmt *r
 	if ru = rcontext.Rule(ctx); ru == nil {
 		return nil, errors.WithStack(errNoRuleFound)
 	}
+
+	var originLimit = new(rast.LimitNode)
+	if stmt.Limit != nil && stmt.Limit.Limit() != 0 {
+		offset := stmt.Limit.Offset()
+		limit := stmt.Limit.Limit()
+		originLimit.SetOffset(offset)
+		originLimit.SetLimit(limit)
+		if stmt.Limit.HasOffset() {
+			originLimit.SetHasOffset()
+		}
+
+		stmt.Limit.SetOffset(0)
+		stmt.Limit.SetLimit(offset + limit)
```

Comment:
Consider limit over sql params, for example, `select * from student where ... limit ?,?`